xml/src/main/scala/akka/stream/alpakka/xml/impl/StreamingXmlParser.scala

```diff
@@ -28,8 +29,9 @@ import scala.annotation.tailrec
 
       import javax.xml.stream.XMLStreamConstants
 
-      private val feeder: AsyncXMLInputFactory = new InputFactoryImpl()
-      private val parser: AsyncXMLStreamReader[AsyncByteArrayFeeder] = feeder.createAsyncFor(Array.empty)
+      private val factory: AsyncXMLInputFactory = new InputFactoryImpl()
+      configureFactory(factory)
+      private val parser: AsyncXMLStreamReader[AsyncByteArrayFeeder] = factory.createAsyncFor(Array.empty)
       if (ignoreInvalidChars) {
         parser.getConfig.setIllegalCharHandler(new ReplacingIllegalCharHandler(0))
       }
```

Comment:
While you're at it, would it make sense to allow for parser configuration with the same pattern?

Reply:
not sure there's any other configuration settings for the parser itself that couldn't be set in the factory. I'll look into that though...

Reply:
ok, so I can't spend a ton of time on this, but from the list of public methods available on the ReaderConfig instance (returned from parser.getConfig) that mutate the configuration, they all fall into one of these categories:
1) it's a configuration that could be set on the factory prior to creation of the parser
2) it's something that, from what I can tell, should perhaps not be public at all
3) it's the setIllegalCharHandler() config

1 is now implemented with what we have so far in this pull request
2, it's possible that one or two of the methods I put into this category ARE in fact useful, but since that's not clear, and since there's no current ask from anyone, I would say adding support for these into alpakka is not necessary at this time.
3, this is only partially supported in alpakka right now. Alpakka creates a specific IllegalCharHandler now, whereas the underlying config would allow the use of any custom handler. It might be useful to address this, but it might require a more significant impact on the signature of the parser constructors, in order to work nicely with the existing ignoreInvalidChars parameter. So I'm inclined to not take up that effort with this pull-request.